skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -396,42 +388,36 @@ public function executeIndex( $params ) {
 		$this->loginClass = !empty( $this->wg->DisableAnonymousEditing ) ? ' require-login' : '';
 
 		// render monetization module
-		if ( !empty( $params['monetizationModules'] ) ) {
-			$this->monetizationModules = $params['monetizationModules'];
-		}
+		$this->monetizationModules = $this->request->getArray( 'monetizationModules' );
 
 		wfProfileOut( __METHOD__ );
 	}
 
 	/**
 	 * Render header for edit page
 	 */
-	public function executeEditPage() {
-		global $wgTitle, $wgRequest, $wgSuppressToolbar, $wgShowMyToolsOnly, $wgEnableWallExt;
-		wfProfileIn( __METHOD__ );
-
+	public function editPage() {
 		// special handling for special pages (CreateBlogPost, CreatePage)
-		$ns = $wgTitle->getNamespace();
-		if ( $ns == NS_SPECIAL ) {
-			wfProfileOut( __METHOD__ );
+		if ( $this->wg->Title->isSpecialPage() ) {
 			return;
 		}
+		wfProfileIn( __METHOD__ );
 
 		// detect section edit
-		$isSectionEdit = is_numeric( $wgRequest->getVal( 'section' ) );
+		$isSectionEdit = is_numeric( $this->wg->Request->getVal( 'section' ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)